shared/data-scripts/user-data-server.sh

```diff
@@ -51,6 +51,15 @@ consul acl role create -name "nomad-auto-join" -description "Role with policies
 
 consul acl token create -accessor=${nomad_consul_token_id} -secret=${nomad_consul_token_secret} -description "Nomad server/client auto-join token" -role-name nomad-auto-join -token-file=$CONSUL_BOOTSTRAP_TOKEN
 
+# Update anonymous token policy to support DNS lookup against Consul DNS
+consul acl policy create \
+    -name anonymous-policy \
+    -rules 'service_prefix "" { policy = "read" } node_prefix "" { policy = "read" }' \
+    -token-file $CONSUL_BOOTSTRAP_TOKEN
+
+consul acl token update -id anonymous -policy-name anonymous-policy -token-file $CONSUL_BOOTSTRAP_TOKEN
```

Comment:
@Ranjandas we're not doing any DNS lookups with consul as part of the cluster setup - is this meant for user's services that may be running on the cluster post-setup? 

Adding the permissions to the anonymous policy exposes the service/node info to anyone with the URL and is not a thing we want to do so I just want to understand what the intention was here.

Reply:
I suppose there should be an option to allow DNS queries only from private IPs.

Reply:
Hi @tunzor, @im2nguyen, the service discovery in Nomad is by querying the API using templating right? (As shown in this tutorial https://developer.hashicorp.com/nomad/tutorials/service-discovery/service-discovery-app-deployment), and not by native DNS query. This change is to allow two services talk to each other purely using Consul DNS Names. Please let me know if this makes sense.

Reply:
Ahh I understand what I've missed now, my mistake for conflating the two. I was testing with jobs that use both Nomad and Consul service discovery.

I'm going to talk through these changes more with the education team and see if we can have Consul DNS accessible without over-privileging the anonymous token and not adding too much more to the tutorials.